impl/authentication/fake_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authentication
+
+import (
+	"context"
+	"testing"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestBasicValidateCreds(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		desc      string
+		ctx       context.Context
+		wantEmail string
+		wantCode  codes.Code
+	}{
+		{desc: "missing authentication", ctx: ctx, wantCode: codes.Unauthenticated},
+		{desc: "working case", ctx: WithOutgoingFakeAuth(ctx, "foo"), wantEmail: "foo"},
```

Comment:
Aren't there more than two cases e.g. corrupt token? There are multiple errors defined below, which makes me think so.

Reply:
`FakeAuthFunc` is fairly simple.  I'm getting 100% code coverage.

Reply:
OK we shouldn't really be testing fakes though. Are there tests for the multiple possible errors defined or is that elsewhere?

Reply:
`TestGoogleAuthn` tests those code paths.